crates/components/src/filesystem_view.rs

```diff
@@ -0,0 +1,718 @@
+                        Entry::Dir {
+                            name: subentry_name,
+                            selected,
+                            initialized: false,
+                            depersisted: false,
+                            expanded: false,
+                            total_children: 0,
+                            selected_children: 0,
+                            partial_children: 0,
+                        },
+                        &mut self.arena,
+                    );
+                    if should_select {
+                        self.select(child_id);
+                    }
+                }
+            }
+        }
+
+        if let Some(length) = length {
+            if let Entry::Dir {
+                selected,
+                total_children,
+                selected_children,
+                ..
+            } = self.arena[node_id].get_mut()
+            {
+                *total_children = length;
+                if *selected {
+                    *selected_children = length;
+                }
+            }
+        }
+
+        let mut should_toggle = false;
+
+        let mut frame = egui::containers::Frame::none();
+        if self.row_index % 2 != 0 {
+            frame = frame.fill(ui.visuals().faint_bg_color);
+        }
+        self.row_index += 1;
+
+        let mut header_response = None;
+
+        match self.arena[node_id].get_mut() {
+            Entry::File { name, selected } => {
+                frame.show(ui, |ui| {
+                    if ui
+                        .add(egui::SelectableLabel::new(*selected, name.to_string()))
+                        .clicked()
+                    {
+                        should_toggle = true;
+                    };
+                });
+            }
+            Entry::Dir {
+                depersisted,
+                selected,
+                expanded,
+                selected_children,
+                partial_children,
+                ..
+            } => {
+                let id = self.id.with(node_id);
+
+                // De-persist state of the collapsing headers since the underlying filesystem may
+                // have changed since this view was last used
+                if !*depersisted {
+                    *depersisted = true;
+                    if let Some(h) = egui::collapsing_header::CollapsingState::load(ui.ctx(), id) {
+                        h.remove(ui.ctx())
+                    }
+                    ui.ctx().animate_bool_with_time(id, *expanded, 0.);
+                }
+
+                let header = egui::collapsing_header::CollapsingState::load_with_default_open(
+                    ui.ctx(),
+                    id,
+                    *expanded,
+                );
+
+                *expanded = header.openness(ui.ctx()) >= 0.2;
+
+                let layout = *ui.layout();
+                header_response = Some(header.show_header(ui, |ui| {
```

Comment:
It doesn't seem possible to select different files/folders without selecting the folders inbetween?
![luminol_wOq0Hzwpcg](https://github.com/Astrabit-ST/Luminol/assets/54482069/381e60cc-8c7c-4e0d-8ae6-748cac2b7796)

Reply:
Managed to get Fading Memory running (mostly) off of an rgssad produced by Luminol. I was pleasantly surprised by this, as RPG Maker fails to create archives for games > 128mb!

Reply:
You can select multiple entry by using Ctrl+click on each.

You can select more than one range by using Ctrl+click to select the first endpoint of each range and Ctrl+Shift+click to select the second endpoint.

You can also perform negative selection by first selecting a range, then using Ctrl+click to deselect one endpoint of the negative selection inside of the first selection and Ctrl+Shift+click on the second endpoint of the negative selection.

Reply:
Could you add a tooltip explaining that?

Reply:
Sure, although these are exactly the semantics from GTK's file picker

Reply:
Oh. I honestly didn't know that

Looks like it even applies to Windows Explorer 🤦‍♀️

Reply:
I added a tooltip that shows up when hovering over the filesystem view, and it hides once the user selects something